src/integrationTest/java/uk/gov/hmcts/reform/cpo/wiremock/extension/DynamicOAuthJwkSetResponseTransformer.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.cpo.wiremock.extension;
+
+import com.github.tomakehurst.wiremock.extension.Parameters;
+import com.github.tomakehurst.wiremock.http.Request;
+import com.github.tomakehurst.wiremock.http.Response;
+import com.nimbusds.jose.JOSEException;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.hmcts.reform.cpo.utils.KeyGenUtil;
+
+/*
+ * Replaces response body with the OAuth JWK Set, i.e. the public keys used to sign the mock OAAuth token
+ */
+@Slf4j
+public class DynamicOAuthJwkSetResponseTransformer extends AbstractDynamicResponseTransformer {
+
+    static final String DYNAMIC_OAUTH_JWK_SET_RESPONSE_TRANSFORMER = "dynamic-oauth-jwk-set-response-transformer";
+
+    @Override
+    protected String dynamicResponse(Request request, Response response, Parameters parameters) {
+
+        try {
+            return "{"
+                + "   \"keys\": [" +  KeyGenUtil.getRsaJWK().toJSONObject() + "]"
```

Comment:
Seems like a static one. Can't we set in the wiremock json mapping itself?

Reply:
Yes, response would be static for all calls (assuming key signature doesn't have an expiry date) so it could have hard coded as per test stub service (having borrowed other elements of test stub service, I haddn't noticed that stub file).  

However **leaving as is** as it means we would only need to change `KeyGenUtil` if for any reason we needed to randomise the test key_id or increase the key strength in the future.